File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/ChangePassword.java
package l2s.gameserver.network.authcomm.gs2as;

import io.netty.buffer.ByteBuf;
import l2s.gameserver.network.authcomm.SendablePacket;

public class ChangePassword extends SendablePacket {
    private final String _account;
    private final String _oldPass;
    private final String _newPass;
    private final String _hwid;

    public ChangePassword(String account, String oldPass, String newPass, String hwid) {
        _account = account;
        _oldPass = oldPass;
        _newPass = newPass;
        _hwid = hwid;
    }

    @Override
    protected ByteBuf writeImpl(ByteBuf byteBuf) {
        writeString(byteBuf, _account);
        writeString(byteBuf, _oldPass);
        writeString(byteBuf, _newPass);
        writeString(byteBuf, _hwid);
        return byteBuf;
    }

    @Override
    protected byte getOpCode() {
        return 8;
    }
}

File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/ChangePassword.kt
package l2s.gameserver.network.authcomm.gs2as

import io.netty.buffer.ByteBuf
import l2s.gameserver.network.authcomm.SendablePacket

/**
 * @Author: Death
 * @Date: 8/2/2007
 * @Time: 14:35:35
 */

class ChangePassword(
    private val _account: String,
    private val _oldPass: String,
    private val _newPass: String,
    private val _hwid: String
) : SendablePacket() {

    override val opCode: Byte = 0x08

    override fun writeImpl(byteBuf: ByteBuf): ByteBuf {
        writeString(byteBuf, _account)
        writeString(byteBuf, _oldPass)
        writeString(byteBuf, _newPass)
        writeString(byteBuf, _hwid)

        return byteBuf
    }

}